data_service/adapters/local_storage.py

```diff
@@ -0,0 +1,130 @@
+import json
```

Comment:
Not sure about this. It looks like a lot of code in this file will have to be repeated for a future cloud adapter.

Reply:
As I understand the [Google Cloud storage API](https://cloud.google.com/python/docs/reference/storage/latest)  and as it was implemented before, the code downloaded a file from bucket locally to Docker container, so there were no Docker mounts in that case. A file's path in bucket has another format then local, that's why one needs a local storage adapter that returns the local file path and a bucket adapter that returns the "bucket" file path.

Reply:
Yes. But there would always be an option (if we ever move into the cloud ☁️ ) to just mount actual disks to the cluster and use that. Using buckets exclusively would also mean rewrites to both metadata-service, job-service and job-executor.

To support this flexibility in all components when the move to cloud (and the choice to use buckets rather than file system) is so unclear that it seems like a lot of complexity for possibly no gain 🤷🏻 

That being said it is not a huge deal to move this method out to the domain/data.py and just get the filepath from the local_storage adapter if that is preferable. 🙇🏻

Reply:
> That being said it is not a huge deal to move this method out to the domain/data.py and just get the filepath from the local_storage adapter if that is preferable.

I like this!